app_dart/lib/src/service/flags/dynamic_config.dart

```diff
@@ -0,0 +1,139 @@
+///
+/// To get a one-time copy of the file, see also:
+///
+/// - [DynamicConfig.fromRemoteLatest]
+/// - [DynamicConfig.fromLocalRepo]
+@JsonSerializable(explicitToJson: true)
+@immutable
+final class DynamicConfig {
+  /// Default configuration for flags.
+  static const defaultInstance = DynamicConfig._(
+    backfillerCommitLimit: 50,
+    contentAwareHashing: ContentAwareHashing.defaultInstance,
+  );
+
+  /// Upper limit of commit rows to be backfilled in API call.
+  ///
+  /// This limits the number of commits to be checked to backfill. When bots
+  /// are idle, we hope to scan as many commit rows as possible.
+  @JsonKey()
+  final int backfillerCommitLimit;
+
+  /// Flags associated with content-aware hashing.
+  @JsonKey()
+  final ContentAwareHashing contentAwareHashing;
+
+  const DynamicConfig._({
+    required this.backfillerCommitLimit,
+    required this.contentAwareHashing,
+  });
+
+  /// Creates [DynamicConfig] flags from a [json] object.
+  ///
+  /// Any omitted fields default to the values in [defaultInstance].
+  factory DynamicConfig({
+    int? backfillerCommitLimit,
+    ContentAwareHashing? contentAwareHashing,
+  }) {
+    return DynamicConfig._(
+      backfillerCommitLimit:
+          backfillerCommitLimit ?? defaultInstance.backfillerCommitLimit,
+      contentAwareHashing:
+          contentAwareHashing ?? defaultInstance.contentAwareHashing,
+    );
+  }
+
+  /// Creates [DynamicConfig] flags from a [json] object.
+  ///
+  /// Any omitted fields default to the values in [defaultInstance].
+  factory DynamicConfig.fromJson(Map<String, Object?>? json) {
+    return _$DynamicConfigFromJson(json ?? {});
+  }
+
+  /// Creates [DynamicConfig] flags from a [yaml] object.
+  factory DynamicConfig.fromYaml(YamlMap? yaml) {
+    return DynamicConfig.fromJson(yaml?.asMap);
+  }
+
+  /// Returns the latest copy of [DynamicConfig] fetched from tip-of-tree.
+  ///
+  /// Equivalent to a single call to [DynamicConfigUpdater.fetchDynamicConfig].
+  static Future<DynamicConfig> fromRemoteLatest() async {
+    return DynamicConfigUpdater().fetchDynamicConfig();
+  }
+
+  /// Returns the latest copy of [DynamicConfig] fetched from the repository.
+  static Future<DynamicConfig> fromLocalRepo() async {
+    final execPath = io.Platform.resolvedExecutable;
+
+    // Walk backwards until the root of the Cocoon repository is found.
+    var dir = io.File(execPath).parent;
+    while (dir.path != dir.parent.path) {
+      final gitDir = io.Directory(p.join(dir.path, '.git'));
+      if (await gitDir.exists()) {
+        break;
+      }
+      dir = dir.parent;
+    }
+
+    final configPath = io.File(p.join(dir.path, 'app_dart', 'config.yaml'));
+    if (!await configPath.exists()) {
+      throw StateError('Could not find config.yaml at ${configPath.path}');
+    }
+
+    final yaml = loadYaml(await configPath.readAsString()) as YamlMap;
+    return DynamicConfig.fromJson(yaml.cast());
```

Comment:
this will fail because cast<>() doesn't do a deep copy, you'll just hit another YamlMap under the hood as a runtime failure. Call fromYaml()?

Reply:
Doh! I wrote it specifically to avoid doing this and then ... didn't use it. Done!